tests/bson/bson-objectid-getTimestamp-002.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+MongoDB\BSON\ObjectId::getTimestamp: Ensure that the Timestamp field is represented as an unsigned 32-bit integer
+--FILE--
+<?php
+date_default_timezone_set( "UTC" );
+
+function create_object_id($oid_string)
+{
+    $objectid = new MongoDB\BSON\ObjectId($oid_string);
+    $ts = $objectid->getTimestamp();
+    echo $ts, "\n";
+    echo date_create( "@{$ts}" )->format( "Y-m-d H:i:s" ), "\n";
+}
+
+create_object_id('000000000000000000000000');
+create_object_id('7FFFFFFF0000000000000000');
+create_object_id('800000000000000000000000');
+create_object_id('FFFFFFFF0000000000000000');
+
+?>
+--EXPECT--
+0
+1970-01-01 00:00:00
+2147483647
+2038-01-19 03:14:07
+2147483648
+2038-01-19 03:14:08
+4294967295
```

Comment:
I think this will fail on 32-bit platforms, since it will render as a negative value.

https://www.php.net/manual/en/mongodb-bson-objectid.gettimestamp.php talks about this and suggests using `%u` with `sprintf()` to dump the value. That said, I'm not sure how `date_create()` will handle this on a 32-bit platform.

You can try only making the `%u` change (rather than `echo $ts`) and see if this passes on our 32-bit environments in Evergreen.

If that doesn't work (or you'd rather not bother), we can just require a 64-bit platform for this test:

```
--SKIPIF--
<?php if (PHP_INT_SIZE !== 8) { die("skip Difficulty creating dates from unsigned 32-bit ints on a 32-bit platform"); } ?>
```

Reply:
I'm taking the easy way and skip the test on 32-bit platforms: since the logic is entirely in libmongoc, I mainly added these tests as a safeguard. Since libmongoc is tested on i386 platforms I feel safe skipping these tests here.